rqd/rqd/rqnetwork.py

```diff
@@ -206,17 +207,43 @@ def connectGrpcWithRetries(self):
         while True:
             try:
                 self.rqCore.grpcConnected()
+                log.info("Successfully connected to Cuebot.")
+                self.reconnection_attempts = 0
                 break
             except grpc.RpcError as exc:
-                # pylint: disable=no-member
+                # Log the gRPC connection issue
                 if exc.code() == grpc.StatusCode.UNAVAILABLE:
                     log.warning(
-                        'GRPC connection failed. Retrying in %s seconds',
-                        rqd.rqconstants.RQD_GRPC_CONNECTION_ATTEMPT_SLEEP_SEC)
+                        'GRPC connection failed. Retrying in %s seconds (attempt %d)',
+                        rqd.rqconstants.RQD_GRPC_CONNECTION_ATTEMPT_SLEEP_SEC,
+                        self.reconnection_attempts + 1)
+
+                    # Increment reconnection attempts counter
+                    self.reconnection_attempts += 1
+
+                    if self.reconnection_attempts >= rqd.rqconstants.RQD_GRPC_MAX_RETRIES:
+                        log.error("Exceeded maximum reconnection attempts. Marking RQD as wedged.")
+                        self.handle_wedged_state()
+                        break
+
+                    # Sleep before retrying
                     time.sleep(rqd.rqconstants.RQD_GRPC_CONNECTION_ATTEMPT_SLEEP_SEC)
                 else:
+                    log.error("GRPC encountered an unknown error: %s", exc)
                     raise exc
-                # pylint: enable=no-member
+
+    def handle_wedged_state(self):
+        """Handles the RQD wedged state when reconnection attempts are exhausted."""
+        log.error("RQD has entered a wedged state. Triggering automatic restart.")
+        self.restart_rqd()
```

Comment:
Handling a wedged state should simply terminate the process cleanly to give the watchdog a chance to start the process. Having a process restarting itself as a service is not a good approach. Besides that, it can lead to a situation where both this proc and the watchdog proc are trying to restart the service.

Reply:
Done!